providers/coco.go

```diff
@@ -0,0 +1,93 @@
+package providers
+
+import (
+	"fmt"
+	"net/http"
+	"net/url"
+        "errors"
+        "github.com/oauth2-proxy/oauth2-proxy/pkg/apis/sessions"
+        "github.com/oauth2-proxy/oauth2-proxy/pkg/requests"
+)
+
+type COCOProvider struct {
+	*ProviderData
+}
+
+//<!--COCO Begin-->
+//COCO: NewCOCOProvider initiates a new COCOProvider
+//<!--COCO End-->
+func NewCOCOProvider(p *ProviderData) *COCOProvider {
+	p.ProviderName = "COCO"
+	if p.LoginURL.String() == "" {
+		p.LoginURL = &url.URL{Scheme: "https",
+			Host: "api.getcoco.buzz",
```

Comment:
- check if this can be used as env variable

Reply:
we can do ... we can pass as env variable while running the proxy server